tests/ImageSharp.Tests/Formats/Pbm/PbmEncoderTests.cs

```diff
@@ -0,0 +1,144 @@
+                { BlackAndWhiteBinary, PbmColorType.BlackAndWhite },
+                { BlackAndWhitePlain, PbmColorType.BlackAndWhite },
+                { GrayscaleBinary, PbmColorType.Grayscale },
+                { GrayscaleBinaryWide, PbmColorType.Grayscale },
+                { GrayscalePlain, PbmColorType.Grayscale },
+                { RgbBinary, PbmColorType.Rgb },
+                { RgbPlain, PbmColorType.Rgb },
+            };
+
+        [Theory]
+        [MemberData(nameof(PbmColorTypeFiles))]
+        public void PbmEncoder_PreserveColorType(string imagePath, PbmColorType pbmColorType)
+        {
+            var options = new PbmEncoder();
+
+            var testFile = TestFile.Create(imagePath);
+            using (Image<Rgba32> input = testFile.CreateRgba32Image())
+            {
+                using (var memStream = new MemoryStream())
+                {
+                    input.Save(memStream, options);
+                    memStream.Position = 0;
+                    using (var output = Image.Load<Rgba32>(memStream))
+                    {
+                        PbmMetadata meta = output.Metadata.GetPbmMetadata();
+                        Assert.Equal(pbmColorType, meta.ColorType);
+                    }
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(PbmColorTypeFiles))]
+        public void PbmEncoder_WithPlainEncoding_PreserveBitsPerPixel(string imagePath, PbmColorType pbmColorType)
+        {
+            var options = new PbmEncoder()
+            {
+                Encoding = PbmEncoding.Plain
+            };
+
+            TestFile testFile = TestFile.Create(imagePath);
+            using (Image<Rgba32> input = testFile.CreateRgba32Image())
+            {
+                using (var memStream = new MemoryStream())
+                {
+                    input.Save(memStream, options);
+                    memStream.Position = 0;
+                    using (var output = Image.Load<Rgba32>(memStream))
+                    {
+                        PbmMetadata meta = output.Metadata.GetPbmMetadata();
+                        Assert.Equal(pbmColorType, meta.ColorType);
+                    }
+                }
+            }
+        }
+
+        [Theory]
+        [WithFile(BlackAndWhitePlain, PixelTypes.Rgb24)]
+        public void PbmEncoder_P1_Works<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : unmanaged, IPixel<TPixel> => TestPbmEncoderCore(provider, PbmColorType.BlackAndWhite, PbmEncoding.Plain);
+
+        [Theory]
+        [WithFile(BlackAndWhiteBinary, PixelTypes.Rgb24)]
+        public void PbmEncoder_P4_Works<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : unmanaged, IPixel<TPixel> => TestPbmEncoderCore(provider, PbmColorType.BlackAndWhite, PbmEncoding.Binary);
+
+        /* Disabled as Magick throws an error reading the input image
+        [Theory]
+        [WithFile(GrayscalePlain, PixelTypes.Rgb24)]
```

Comment:
Might worth to also feed this to `PbmColorImageCanRoundTrip` since normal don't work on it.

Reply:
Indeed, added the Grayscale images